plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/AbstractXmlSourceValidator.java

```diff
@@ -44,15 +55,21 @@
       AbstractXmlSourceValidator.class.getName());
   
   private IDocument document;
+  private boolean hasValidated;
   
   /**
-   * Validates a given {@link IDocument}.
+   * Validates a given {@link IDocument} if the project has the App Engine Standard facet.
    */
   public void validate(IValidationContext helper, IReporter reporter) throws ValidationException {
+    IProject project = getProject(helper);
     try {
-      String encoding = getDocumentEncoding(document);
-      byte[] bytes = document.get().getBytes(encoding);
-      this.validate(reporter, bytes);
+      IFacetedProject facetedProject = ProjectFacetsManager.create(project);
+      if (AppEngineStandardFacet.hasAppEngineFacet(facetedProject)) {
+        hasValidated = true;
```

Comment:
not sure it matters, but I'd expect this to come after this.validate in case validate() throws an exception